x/ibc/23-commitment/types/merkle.go

```diff
@@ -118,34 +131,50 @@ func (mp MerklePath) Pretty() string {
 
 // IsEmpty returns true if the path is empty
 func (mp MerklePath) IsEmpty() bool {
-	return len(mp.KeyPath) == 0
+	return len(mp.KeyPaths) == 0
 }
 
 // ApplyPrefix constructs a new commitment path from the arguments. It interprets
 // the path argument in the context of the prefix argument.
 //
+// ApplyPrefix will prepend a new Merkle KeyPath to the existing KeyPaths. This
+// allows a single path to be represent a chained merkle proof
+//
 // CONTRACT: provided path string MUST be a well formated path. See ICS24 for
 // reference.
-func ApplyPrefix(prefix exported.Prefix, path string) (MerklePath, error) {
-	err := host.DefaultPathValidator(path)
+func ApplyPrefix(prefix exported.Prefix, path exported.Path) (exported.Path, error) {
+	err := host.DefaultPathValidator(path.String())
 	if err != nil {
 		return MerklePath{}, err
 	}
 
 	if prefix == nil || prefix.IsEmpty() {
 		return MerklePath{}, errors.New("prefix can't be empty")
 	}
-	return NewMerklePath([]string{string(prefix.Bytes()), path}), nil
+	mpath, ok := path.(MerklePath)
```

Comment:
If you require the path to be a `MerklePath` here (runtime) but allow any `exported.Path` interface at compile time (function argument) you open up to annoying runtime failures. Why not just require the concrete type. Especially the return type is often much better as a struct (general Golang rule - accept interfaces, return structs).

I would just export MerklePath.